### PR TITLE
LSNBLDR-877 use Hibernate and siteService to retrieve instead of bad custom SQL

### PR DESCRIPTION
**Pull request by ottenhoff**

### CODE REVIEW

File: lessonbuilder/components/src/java/org/sakaiproject/lessonbuildertool/model/SimplePageToolDaoImpl.java [hunk 1]
```diff
@@ -499,6 +522,21 @@ public SimplePageItem findTopLevelPageItemBySakaiId(String id) {
...
 
+	public List<SimplePageItem> findTopLevelPageItemBySakaiIds(List<String> ids) {
+        DetachedCriteria d = DetachedCriteria.forClass(SimplePageItem.class)
+        	.add(Restrictions.in("sakaiId", ids))
+	    .add(Restrictions.eq("pageId", 0L))
+	    .add(Restrictions.eq("type",SimplePageItem.PAGE));
+
+		List<SimplePageItem> list = (List<SimplePageItem>) getHibernateTemplate().findByCriteria(d);
+
+		if (list == null || list.size() < 1) {
```
Comment (bjones86) [inline lessonbuilder/components/src/java/org/sakaiproject/lessonbuildertool/model/SimplePageToolDaoImpl.java]: And here

File: lessonbuilder/components/src/java/org/sakaiproject/lessonbuildertool/model/SimplePageToolDaoImpl.java [hunk 2]
```diff
@@ -209,19 +212,39 @@ public void clear() {
...
+				sitePageIds.add(page.getId());
+			}
+
+			DetachedCriteria d = DetachedCriteria.forClass(SimplePage.class);
+			d.add(Restrictions.in("toolId", sitePageIds));
+			d.add(Restrictions.isNull("parent"));
+
+			List<SimplePage> lessonsPages = (List<SimplePage>) getHibernateTemplate().findByCriteria(d);
+			List<String> lessonsPageIds = new ArrayList<>();
+			if (lessonsPages != null && lessonsPages.size() > 0) {
```
Comment (bjones86) [inline lessonbuilder/components/src/java/org/sakaiproject/lessonbuildertool/model/SimplePageToolDaoImpl.java]: You could use CollectionUtils.isEmpty(list) here

**Comment by ottenhoff** [comment]
@bjones86 do you have 5million plus rows in lesson_builder_items?

How does Oracle handle that query across two data types (a bigint against a varchar(250)?

File: lessonbuilder/components/src/java/org/sakaiproject/lessonbuildertool/model/SimplePageToolDaoImpl.java (hunk 2 above)
Comment (buckett) [reply lessonbuilder/components/src/java/org/sakaiproject/lessonbuildertool/model/SimplePageToolDaoImpl.java]: Or just `!lessonsPages.isEmpty()` as I don't think `findByCriteria`should return `null`

**Comment by ottenhoff** [comment]
Thanks @buckett. Do you have millions of rows in these tables? Any chance you can partially replicate?

**Comment by bjones86** [comment]
Yeah we have ~20 million in items and ~8 million in pages. We haven't had any reports of this query being slow or not returning, but I unfortunately don't have access to our Oracle OEM software that does this reporting, so we rely on our DBAs/sysadmins to relay this information to us (when and if they identify it).

**Comment by ottenhoff** [comment]
Thanks @bjones86. It would be great if I could get confirmation from someone else who can profile this query on their MySQL before merging

**Pull request merged**